### PR TITLE
Support Decimals

### PR DESCRIPTION
**Pull request by ponimas**
The Decimal datatype is crucial for financial applications hence we have to support it.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ The project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html)
 - `Field`, `CryptoHash`, `StorageValue` and `ExonumJson` traits have been
   implemented for `chrono::Duration` structure. (#653)
 
+- `Field`, `CryptoHash`, `StorageValue` and `ExonumJson` traits have been
+  implemented for `rust_decimal::Decimal`.
```
Comment (vldm) [inline CHANGELOG.md]: Add Pr number pls

Comment (ponimas) [reply CHANGELOG.md]: @vldm fxd

File: exonum/src/storage/values.rs [hunk 2]
```diff
@@ -321,9 +322,23 @@ impl StorageValue for Uuid {
     }
 }
 
+impl StorageValue for Decimal {
+    fn into_bytes(self) -> Vec<u8> {
+        self.serialize().to_vec()
+    }
+
+    fn from_bytes(value: Cow<[u8]>) -> Self {
+        assert_eq!(value.len(), 16);
```
Comment (vitvakatu) [inline exonum/src/storage/values.rs]: You don't need this check

File: exonum/src/storage/values.rs [hunk 3]
```diff
@@ -494,4 +509,21 @@ mod tests {
...
+
+    #[test]
+    fn decimal_round_trip() {
+        let values = [
+            Decimal::from_str("3.14").unwrap(),
+            Decimal::from_parts(1102470952, 185874565, 1703060790, false, 28),
+            Decimal::new(9497628354687268, 12),
+        ];
+
+        for value in values.iter() {
```
Comment (vitvakatu) [inline exonum/src/storage/values.rs]: side note: I wonder why not clippy asking for replacing this with `for value in &values`

File: exonum/src/storage/values.rs [hunk 4]
```diff
@@ -494,4 +509,21 @@ mod tests {
...
+            let bytes = value.clone().into_bytes();
```
Comment (vitvakatu) [inline exonum/src/storage/values.rs]: also side note: you can use `for value in values.into_iter()` and then you don't need `clone()` here

**Comment by ponimas** [comment]
@vitvakatu fxd

File: exonum/src/storage/values.rs [hunk 5]
```diff
@@ -321,9 +322,22 @@ impl StorageValue for Uuid {
     }
 }
 
+impl StorageValue for Decimal {
```
[the remark this answers is not in the document]
Comment (ponimas) [reply exonum/src/storage/values.rs]: IMO I don't think that it's necessary.
But if you insist.. huh

[the remark this answers is not in the document]
Comment (ponimas) [reply exonum/src/storage/values.rs]: @DarkEld3r  done

**Pull request merged**